include/quda_backend.h

```diff
@@ -0,0 +1,139 @@
+#pragma once
```

Comment:
This file should be split in two I think, `quda_target_cuda.h` and `quda_target_hip.h`.  This file, which I would like renamed to `quda_target.h` would then conditionally include one of the other.  Else as more targets are added, this include file will explode.